First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request contains the first commits for this repo, which creates a GitHub Action to apply the standard permissions monitoring and runner hardening that we do at the beginning of every GitHub Actions job.

## 💭 Motivation and context ##

We can use this to DRY out the GitHub Actions workflows in our skeleton repositories.

## 🧪 Testing ##

This change is tested in cisagov/skeleton-generic#??.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Finalize version.

## ✅ Post-merge checklist ##

- [x] Create a release.
- [x] Create `v1.0` and `v1` tags.

jsf9k committed:
Add versioning script and version file

These were copied from cisagov/skeleton-packer.  Also add semver as a
dependency since it is required for the bump-version script.

jsf9k committed:
Add composite GitHub Action that comprises our standard job preamble

This preamble consists of the standard permissions monitoring and
runner hardening that we apply at the beginning of every GitHub
Actions job.

jsf9k committed:
Bump version from 0.0.1 to 1.0.0

jsf9k committed:
Bump version from 1.0.0 to 1.0.0-rc.1

jsf9k committed:
Update README.md boilerplate

jsf9k committed:
Create a (required) input for the actions permission config

dav3r reviewed (approved):
This is good stuff! 🌶️ 💩

mcdonnnj commented on action.yml:
```diff
@@ -0,0 +1,28 @@
+      name: Harden the runner
+      uses: step-security/harden-runner@v2
+      with:
+        egress-policy: audit
```
I think this should also be an input and we can configure it to have `audit` as a default value if preferred.

jsf9k replied:
Please see commit 0cfead5.

mcdonnnj reviewed (approved):
Good idea 🧠. I have one suggestion for your consideration.

jsf9k committed:
Improve actions permission config input description

jsf9k committed:
Add action input for runner hardening egress policy

Co-authored-by: Nick M <50747025+mcdonnnj@users.noreply.github.com>

jsf9k committed:
Change name of permissions monitoring config input

Also:
- Update the description of this input.
- Give the input the same default value that the
GitHubSecurityLab/actions-permission/monitor action uses.
- Make the input not required.

jsf9k committed:
Improve description of egress policy input

jsf9k committed:
Ensure that all steps have an id and a name
